src/core/algorithms/dc/FastADC/util/evidence_aux_structures_builder.h

```diff
@@ -0,0 +1,148 @@
+          eq_mask(0),
+          gt_mask(0) {
+        eq_mask.set(eq_pos);
+        gt_mask.set(gt_pos);
+    }
+};
+
+struct PredicatePacks {
+    /** String single-column predicate packs */
+    std::vector<PredicatePack> str_single;
+    /** String cross-column predicate packs */
+    std::vector<PredicatePack> str_cross;
+    /** Numerical single-column predicate packs */
+    std::vector<PredicatePack> num_single;
+    /** Numerical cross-column predicate packs */
+    std::vector<PredicatePack> num_cross;
+};
+
+/**
+ * Class with structures that helps to build an clue/evidence set.
+ *
+ * This class builds predicate packs, correction map and cardinality mask,
+ * which will be used later to build clue and evidence sets
+ *
+ * A clue set is a compact representation of all satisfied predicates between
+ * tuple pairs in a relation. The clues are built by evaluating the relationships
+ * between tuples using Position List Index (PLI) data structures, and mapping
+ * predicate results to bitsets.
+ */
+class EvidenceAuxStructuresBuilder {
+public:
+    EvidenceAuxStructuresBuilder(EvidenceAuxStructuresBuilder const& other) = delete;
+    EvidenceAuxStructuresBuilder& operator=(EvidenceAuxStructuresBuilder const& other) = delete;
+    EvidenceAuxStructuresBuilder(EvidenceAuxStructuresBuilder&& other) noexcept = default;
+    EvidenceAuxStructuresBuilder& operator=(EvidenceAuxStructuresBuilder&& other) noexcept = delete;
+
+    EvidenceAuxStructuresBuilder(PredicateBuilder const& pbuilder)
+        : num_single_(pbuilder.GetNumSingleColumnPredicates()),
+          num_cross_(pbuilder.GetNumCrossColumnPredicates()),
+          str_single_(pbuilder.GetStrSingleColumnPredicates()),
+          str_cross_(pbuilder.GetStrCrossColumnPredicates()),
+          provider_(pbuilder.predicate_index_provider) {}
+
+    /* For tests */
+    size_t GetNumberOfBitsInClue() const {
+        return correction_map_.size();
+    }
+
+    /* For tests */
+    std::vector<PredicatePack> GetPredicatePacksAsOneVector() const {
+        std::vector<PredicatePack> ret;
+        ret.insert(ret.end(), packs_.str_single.begin(), packs_.str_single.end());
+        ret.insert(ret.end(), packs_.str_cross.begin(), packs_.str_cross.end());
+        ret.insert(ret.end(), packs_.num_single.begin(), packs_.num_single.end());
+        ret.insert(ret.end(), packs_.num_cross.begin(), packs_.num_cross.end());
+        return ret;
+    }
+
+    std::vector<PredicateBitset> const& GetCorrectionMap() const {
+        return correction_map_;
+    }
+
+    PredicatePacks const& GetPredicatePacks() const {
+        return packs_;
+    }
+
+    PredicateBitset const& GetCardinalityMask() const {
+        return cardinality_mask_;
+    }
+
+    /**
+     * Builds predicate packs, correction map and cardinality mask
+     */
+    void BuildAll();
+
+private:
+    PredicateBitset BuildMask(PredicatesSpan group, std::initializer_list<OperatorType>& types);
```

Comment:
Let's not use std::initializer_list to store anything, it's the type mainly for compiler, not for programmer

Reply:
yep, constexpr arrays will do the job